backend/src/handlers_prelude/github_handlers.rs

```diff
@@ -0,0 +1,236 @@
+use axum::{
+    extract::{State, Path},
+    http::StatusCode,
+    Json, Router,
+};
+use axum::routing::{get, post, put};
+use tracing::{error,info};
+use serde::{Serialize, Deserialize};
+use crate::gh::{get_all_branch_details, create_pull_request};
+use crate::handlers_prelude::eyre_to_axum_err;
+use crate::AppState;
+use color_eyre::Result;
+
+/// General API response structure
+#[derive(Serialize, Debug)]
+pub struct ApiResponse<T> {
+    pub status: String,
+    pub message: String,
+    pub data: Option<T>,
+}
```

Comment:
I *love* the idea of having a generalized response type, that would great organizationally, however I feel it should be `type ApiResponse = Result<T, (StatusCode, String)>`, and be defined in `handlers_prelude/mod.rs`. This would help enforce more consistency and make it so that every handler doesn't need to have the whole result spiel typed out.

If you feel that there's a need to have a separate `data` field, then please push back on this.

`status` should probably be a `StatusCode`, you might need to `impl Serialize for StatusCode`

Reply:
I'm not really sure how to do this, if you could give me some information, I would love it.

Reply:
Don't worry about implementing the whole error type thing in this PR unless you really really want to, this can happen with the giant spring cleaning that will take place after assets/branches

Should you want to, see <https://github.com/r-Techsupport/hyde/issues/253>. This would involve locating every use of `eyre_to_axum_err` and replacing it with `?`

Otherwise, just make the error type for this function `(StatusCode, String)`